File: Daily_ui_objc_set/Daily_ui_objc_set/ViewController.m
//
//  ViewController.m
//  Daily_ui_objc_set
//
//  Created by Jason on 17/12/2016.
//  Copyright © 2016 Jason. All rights reserved.
//

#import "ViewController.h"
#import "ComConfig.h"

@interface ViewController () <UITableViewDelegate,UITableViewDataSource>

@property (nonatomic) NSInteger sections;
@property (nonatomic,strong) UITableView *tableView;
@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
}

- (void)viewWillAppear:(BOOL)animated {
    [super viewWillAppear:animated];
    [self.view addSubview:[self tableView]];
    _tableView.dataSource = self;
    _tableView.delegate = self;
}


- (UITableView *)tableView {
    if (_tableView == nil) {
        _tableView = [[UITableView alloc] initWithFrame:CGRectMake(0, 0, [UIScreen mainScreen].bounds.size.width, [UIScreen mainScreen].bounds.size.height) style:UITableViewStylePlain];
    }
    return _tableView;
}

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
    return [ComConfig sharedConfig].dataList.count;
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:[ComConfig sharedConfig].moduleName];
    if (cell == nil) {
        cell = [[UITableViewCell alloc] initWithStyle:UITableViewCellStyleDefault reuseIdentifier:[ComConfig sharedConfig].moduleName];
    }
    cell.textLabel.text = [ComConfig sharedConfig].dataList[indexPath.row][@"name"];
    return cell;
}

- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath {
    UIViewController *containViewController = [[NSClassFromString([ComConfig sharedConfig].dataList[indexPath.row][@"ctrl"]) alloc] init];
    containViewController.title = [ComConfig sharedConfig].dataList[indexPath.row][@"name"];
    [self.navigationController pushViewController:containViewController animated:true];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


@end

File: Daily_ui_set/Daily_ui_set/ViewController.swift
//
//  ViewController.swift
//  Daily_ui_set
//
//  Created by Jason on 07/12/2016.
//  Copyright © 2016 Jason. All rights reserved.
//

import UIKit

class ViewController: UIViewController {
    open let sections = 1
    @IBOutlet weak var tableView: UITableView!
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
}

extension ViewController : UITableViewDataSource {
    func numberOfSections(in tableView: UITableView) -> Int {
        return sections
    }
    
    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return ComConfig.shared.dataList.count
    }
    
    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        
        var cell = tableView.dequeueReusableCell(withIdentifier: ComConfig.shared.moduleName)
        if let _ = cell {
            cell = UITableViewCell(style: .default, reuseIdentifier: ComConfig.shared.moduleName)
        }
        cell?.textLabel?.text = ComConfig.shared.dataList[indexPath.row]["name"]
        return cell!
    }
}

extension ViewController : UITableViewDelegate {
    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
        
        // http://stackoverflow.com/questions/24030814/swift-language-nsclassfromstring/32265287#32265287
        let vc = ContainerViewController()
        vc.title = (ComConfig.shared.dataList[indexPath.row]["name"])
        let vClsString = ComConfig.shared.moduleName + "." + (ComConfig.shared.dataList[indexPath.row]["view"])! as String
        let v = (NSClassFromString(vClsString) as? UIView.Type)?.init()
        vc.sView = v
        vc.orderId = indexPath.row
        navigationController?.pushViewController(vc, animated: true)
    }
}
